pandapipes/test/api/test_components/test_ext_grid.py

```diff
@@ -22,11 +22,9 @@ def test_ext_grid_sorting(use_numba):
     j5 = pandapipes.create_junction(net, 1, 293.15, index=6)
     j6 = pandapipes.create_junction(net, 1, 293.15, index=7)
 
-    pandapipes.create_ext_grid(net, j2, 1, 285.15, type="pt")
-    pandapipes.create_ext_grid(net, j3, 1, 285.15, type="pt")
-    pandapipes.create_ext_grid(net, j5, 1, 285.15, type="t")
-    pandapipes.create_ext_grid(net, j1, 1, 285.15, type="pt")
-    pandapipes.create_ext_grid(net, j1, 1, 285.15, type="pt")
+    pandapipes.create_ext_grids(net, [j2, j3, j5, j1, j1], p_bar=1, t_k=285.15,
+                                type=["auto", "pt", "t", "pt", "tp"])
+    assert np.all(net.ext_grid.type == ["pt", "pt", "t", "pt", "pt"])
```

Comment:
Maybe you should also add a test, where some ps and ts are None. 

Furthermore, what seems a bit strange to me. If I define p and have a t ext grid, the p wouldn't have any effect. 

Therefore, I would set p and t by default equals to None (see my ext_grid comment). Also there should be some kind of warning. If you set p and say the type is 't', a warning should pop up saying something like p has no effect.

Reply:
I did add the check for non-null p or t values.

Reply:
About the None default, I am not so sure. I thought about that and didn't come to a clear conclusion, as both is not exactly user-friendly: on the one hand, giving many default values and expecting one to be set, on the other hand giving some positional arguments and expecting the user to know that in some cases one of them should be set to None (which would be an argument for a default value).

Reply:
Really difficult question. Maybe we can discuss this again.

Reply:
Solution: For ext_grids, None is used as default for p_bar and t_k, the type-keyword was moved a little further up front. For circulation pumps, the p_bar value must still be given and t_k remains None by default, also here the type keyword was moved a little further up front.